### DISCUSSION TITLE
feature request: Auto executing generated code in OpenWeb UI

### DISCUSSION THREAD

**Discussion by aouicher** [open-webui/open-webui#5678] [General]
Currently, when OpenWeb UI generates code snippets, such as Python code, there is no immediate way to execute the code within the interface. This lack of functionality makes it difficult for users to quickly test and verify the generated code, leading to a less interactive and efficient user experience. Very useful to generate graph e.g.

I would like to see an "Admin Run" button added to the OpenWeb UI interface, specifically for code generation scenarios. The feature should:

Appear only when code (e.g., Python) is generated
Be visible and accessible only to users with appropriate permissions or all users
Execute the code in a secure environment
Display the output of the code execution within the interface

**Comment by Hello-World-Traveler** [comment]
Auto executing generated code when using jupyter would also be a good option
https://docs.openwebui.com/tutorials/integrations/jupyter#what-are-jupyter-notebooks
